src/main/java/com/quantumsoft/qupathcloud/dao/CloudDAOImpl.java

```diff
@@ -68,8 +68,16 @@ public CloudDAOImpl(OAuth20 oAuth20) {
                 .setScheme(SCHEME)
                 .setHost(CLOUD_RESOURCE_MANAGER_HOST)
                 .setPath(PATH_TO_PROJECTS);
-        return createRequestForObjectList(uriBuilder, new TypeReference<Projects>() {
-        }).getProjects();
+        Projects projectsOnPage = createRequestForObjectList(uriBuilder, new TypeReference<Projects>() {});
+        List<Project> projectList = new ArrayList<>(projectsOnPage.getProjects());
+        String nextPageToken = projectsOnPage.getNextPageToken();
+        while (nextPageToken != null) {
```

Comment:
I'm worried this might be too slow if we have to load everything at once... can I get a pre-built JAR with this feature so I can test?

Reply:
I sent link to qupath-cloud.jar to your email. I tested this code with the query parameter pageSize=1, for several projects and it works quite fast.

Reply:
Ok yup you're right it's very fast, all good!